apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ids/DistributedTraceIds.java

```diff
@@ -37,6 +38,13 @@ public void append(DistributedTraceId distributedTraceId) {
         if (relatedGlobalTraces.size() > 0 && relatedGlobalTraces.getFirst() instanceof NewDistributedTraceId) {
             relatedGlobalTraces.removeFirst();
         }
+        /*
+         * There maybe an exception in some plugins, which causes the span to fail to exit,
+         * which results in the ThreadLocal not being cleaned up, and this data is increasing.
+         */
+        if (relatedGlobalTraces.size() >= Config.Agent.TRACE_SEGMENT_REF_LIMIT_PER_SEGMENT) {
```

Comment:
Check the codes of `getRelatedGlobalTraces`, it seems we need one reference here only?